src/modules/claim/claim.controller.ts

```diff
@@ -75,14 +79,30 @@ export class ClaimController {
     @Param('did') did: string,
     @Body() data: IClaimIssuance,
   ) {
+    const didDoc = await this.didService.getById(did);
+    const proofVerifier = new ProofVerifier(didDoc);
+
+    if (
+      data.issuedToken &&
+      !(await proofVerifier.verifyAssertionProof(data.issuedToken))
+    ) {
+      throw new ForbiddenException('User signature not valid');
+    }
+
+    if (
+      data.onChainProof &&
+      !(await proofVerifier.verifyAssertionProof(data.onChainProof))
+    ) {
+      throw new ForbiddenException('User signature not valid');
+    }
```

Comment:
@nichonien @JGiter I think this is incorrect. I don't think that `proofVerifier.verifyAssertionProof` can verify `onChainProof`, right?

Reply:
Ah, right. It is only signature part of jwt expected by `ProofVerifier`. So this check probably can be removed because identity of issuer of onchain proof is verified by Claim Manager